### ISSUE TITLE
Runtime error when running tests with coverage on Github actions

### ISSUE THREAD

**Issue by navinpeiris** [edgurgel/mimic#70]
Hi,

```
Mimic version: 1.8.1

Erlang/OTP 27 [erts-15.0] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [jit:ns]
Elixir 1.17.1 (compiled with Erlang/OTP 27)
```

I'm getting the following error when running tests with coverage on Github actions. You can see the error if you expand the "Verify" section in this github action run in this sample project:

https://github.com/navinpeiris/mimic-elixir-1.17-issue/actions/runs/9630569414/job/26561270358

<img width="733" alt="Screenshot 2024-06-23 at 9 23 39 AM" src="https://github.com/edgurgel/mimic/assets/410939/d763be8e-f287-4350-9e0a-040f7bd94681">

```
IMPORTANT: I'm unable to reproduce this on my local Mac OS machine. Only happens when running on Github Actions.
```

Note: Although the run is successful in this sample project, it gets marked as an error in my source repo.

### Error:

```
Analysis includes data from imported files
["/home/runner/work/mimic-elixir-1.17-issue/mimic-elixir-1.17-issue/Elixir.HelloWorld-1922.coverdata",
 "/home/runner/work/mimic-elixir-1.17-issue/mimic-elixir-1.17-issue/Elixir.HelloWorld.Mimic.Original.Module-1922.coverdata"]
Error: 03:48:30.967 [error] Process #PID<0.2325.0> raised an exception
** (ArgumentError) errors were found at the given arguments:
  * 1st argument: must be one of: function or line
    (kernel 10.0) :code.get_coverage(:cover_id_line, HelloWorld.Mimic.Original.Module)
    (tools 4.0) cover.erl:2367: :cover.native_move/1
    (tools 4.0) cover.erl:2318: :cover.move_counters/2
    (stdlib 6.0) lists.erl:2077: :lists.map/2
    (tools 4.0) cover.erl:[30](https://github.com/navinpeiris/mimic-elixir-1.17-issue/actions/runs/9630569414/job/26561270358#step:5:31)38: anonymous fn/2 in :cover.pmap_spawn/4
Generated HTML coverage results in "cover" directory
```

**Comment by navinpeiris** [comment]
The reason that the run passes in the sample repo and fails on my source repo is because this error caused the coverage to fall below the coverage threshold.

**Comment by edgurgel** [comment]
Thanks for the report! 

The issue seems to only trigger when JIT is enabled and it's not always that I can trigger it. Anyway I think the issue has been fixed with Mimic 1.8.2.

**Comment by navinpeiris** [comment]
Thanks for that @edgurgel, and yes, can confirm that the issue is fixed in Mimic 1.8.2. Thanks a ton for the quick turnaround, especially on a weekend! 🔥
